sprite.go

```diff
@@ -27,18 +27,45 @@ func (s *SpriteSheet) Release() {
 }
 
 // DrawSprite draws a sprite from the sprite sheet at the given position.
-func (s *SpriteSheet) DrawSprite(col, row uint, pos Position, flip bool) {
+func (s *SpriteSheet) DrawSprite(col, row uint, pos Position, scale float32, flip bool) {
```

Comment:
I think the scale should be an integer that indicates the denominator of the scale fraction. So we do not promote the usage of odd dimensions.